packages/router/src/__typetests__/tsconfig.json

```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../../tsconfig.json"
```

Comment:
Try adding the following lines and importing from `.ts` file. Does is help?

```json
"compilerOptions": {
  "composite": false,
}
```

Reply:
Thanks for looking into it! 

Yep I had to set composite, but also set incremental - because the base tsconfig contains `tsBuildInfo` - which needs one of these flags. I'm not 100% sure the impact this has though, but since its only for the tsd test environment should be ok, I think

```
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "composite": false,
    "incremental": true
  },
}
```

Reply:
Yes, the file is included. I see the same config from inside `tsd-lite`. It is using TS compiler’s APIs to [find and read config](https://github.com/mrazauskas/tsd-lite/blob/cbd30a7469b10f2abac2792083cd859b6b2dc3ff/source/utils/resolveCompilerOptions.ts#L8-L17).

The error we saw comes from TS compiler and it looks misleading. Here is an experiment:

1. Add empty `tsconfig.json` to `__typetests__` folder.
2. Run `yarn tsc -p src/__typetests__`. All works.
4. Add `"compilerOptions": {"composite": true}` to the same `tsconfig.json` and run the same command. 
5. Oh.. that’s the error we saw.. And it came from `tsc` this time.
6. Now also add `"references": [{ "path": "../../" }]` to the same `tsconfig.json` and run the command.

All works again. Isn’t it that with `"composite": true` each directory with `tsconfig.json` turns into a project? So in this case the compiler complains, because it sees `src` and `__typetests__` as separate projects.

By the way, `tsd-lite` wraps TS compilers’s APIs to [typecheck and compare types](https://github.com/mrazauskas/tsd-lite/blob/cbd30a7469b10f2abac2792083cd859b6b2dc3ff/source/tsdLite.ts#L30). If `tsc` complains, `tsd-lite` will complain in the same way.

Reply:
Thanks for the explanation Tom, makes sense. 

The TS docs lack a little bit on `composite` and `incremental` - but it certainly seems to behave the way you're describing. I'll close the issue on the jest runner side :)

Reply:
Yep. This was tricky to figure out, but makes sense. It also explains why running test agains `.d.ts` worked for you.

Reply:
Just had a thought @mrazauskas - more for my knowledge than anything else - why does it matter if the typetests folder is treated as a different project (due to the composite flag)? 

In theory the `typetests` project is receiving the same list of files as the actual project right?

Reply:
Wait. I just realised that `tsc` needs `-b` flag to work with composites, not `-p`. So error comes from something else. Hm.. Just tried to use `yarn tsc -b src/__typetests__` with previous experiment. Same result.

Back to your question, `tsd-lite` looks for a `tsconfig.json` nearest to a test file. If not found, it will use defaults of the compiler. If you delete `__typetests__/tsconfig.json`, `tsd-lite` will use config from `packages/router`. It gives the same error, adding `{"composite": false}` to `packages/router/tsconfig.json` makes it work.

Very interesting case. I have to dig deeper into this to understand. In any case, since `tsc` throws the same error this must be a config issue.

Reply:
Can’t stop ;D I just looked at Jest repo where `tsd-lite` is used (with 1200+ assertions passing) and in `__typetests__/tsconfig.json` we have `"noUnusedLocals": false`, `"noUnusedParameters": false`, etc. In a larger test suite these are useful. With time it becomes handy to have dedicated `tsconfig.json` for type tests, although it might look strange at first.

Reply:
I mean to ask a much simpler question, why doesn't tsd-lite work with composite? Just for knowledge - I don't quite grasp what the composite flag is for (outside of a build-time optimisation). 

Thank you very much for looking into it - glad it's piqued your interest!

Reply:
Oh my - I think I spotted the problem! Look at the last exclude glob, it ignores the test files 😅

Reply:
Right. I was looking at that glob. Might be the cause, but note that `routeParamsTypes.ts` is reported as not included, which is not a test file. Also you need those globs for build. So they have to stay.

I think `tsd-lite` should work with composites, but with `__typetests__/tsconfig.json` it needs `references`. Just like `tsc` in my experiment, which I described above. At the same time, `composite` is useful for building libraries, there is not much use of it in testing. Perhaps there is a use case, but I think `"composite": false` is better solution in your case.